spacialgaze-plugins/exp.js

```diff
@@ -0,0 +1,284 @@
+	for (let i = 0; i < benchmarks.length; i++) {
+		if (curExp >= 140000) return "Maxed!";
+		if (benchmarks[i] <= curExp) {
+			continue;
+		} else {
+			return i;
+		}
+	}
+}
+SG.level = level;
+
+function nextLevel(user) {
+	let curExp = Db.exp.get(user, 0);
+	let benchmarks = [0, 40, 90, 165, 250, 400, 600, 810, 1250, 1740, 2450, 3300, 4400, 5550, 6740, 8120, 9630, 11370, 13290, 15520, 18050, 23000, 28000, 33720, 39900, 46440, 52690, 58000, 63600, 69250, 75070, 81170, 87470, 93970, 100810, 107890, 115270, 122960, 131080, 140000];
+	for (let i = 0; i < benchmarks.length; i++) {
+		if (curExp >= 140000) return "no more level ups :(";
+		if (benchmarks[i] <= curExp) {
+			continue;
+		} else {
+			return benchmarks[i] - curExp + " exp";
+		}
+	}
+}
+SG.nextLevel = nextLevel;
+
+exports.commands = {
+	level: 'exp',
+	xp: 'exp',
+	exp: function (target, room, user) {
+		if (!this.runBroadcast()) return;
+		if (!target) target = user.name;
+
+		const targetId = toId(target);
+
+		EXP.readExp(targetId, exp => {
+			this.sendReplyBox('<b>' + SG.nameColor(targetId, true) + '</b> has ' + exp + ' exp and is level ' + SG.level(targetId) + ' and has ' + SG.nextLevel(targetId) + ' until the next level.');
+		});
+	},
+
+	givexp: 'giveexp',
+	giveexp: function (target, room, user) {
+		if (!this.can('roomowner')) return false;
+		if (!target || target.indexOf(',') < 0) return this.parse('/help giveexp');
+
+		let parts = target.split(',');
+		let username = parts[0];
+		let uid = toId(username);
+		let amount = isExp(parts[1]);
+
+		if (amount > 1000) return this.sendReply("You cannot give more than 1,000 exp at a time.");
+		if (username.length >= 19) return this.sendReply("Usernames are required to be less than 19 characters long.");
+		if (typeof amount === 'string') return this.errorReply(amount);
+
+		SG.addExp(uid, this.room, amount);
+		this.sendReply(uid + " has received " + amount + ((amount === 1) ? " exp." : " exp."));
+	},
+	giveexphelp: ["/giveexp [user], [amount] - Give a user a certain amount of exp."],
+
+	resetexp: 'resetxp',
+	confirmresetxp: 'resetxp',
+	resetxp: function (target, room, user, cmd) {
+		if (!target) return this.errorReply('USAGE: /resetxp (USER)');
+		let parts = target.split(',');
+		let targetUser = parts[0].toLowerCase().trim();
+		if (!this.can('roomowner')) return false;
+		if (cmd === 'resetxp' || cmd === 'resetexp') {
+			return this.popupReply('|html|<center><button name="send" value="/confirmresetxp ' + targetUser + '"style="background-color:red;height:300px;width:150px"><b><font color="white" size=3>Confirm XP reset of ' + SG.nameColor(targetUser, true) + '; this is only to be used in emergencies, cannot be undone!</font></b></button>');
+		}
```

Comment:
This did not trigger when i ran `/resetexp`. Check to make sure `cmd` is the right value (wrong order probably effects it, connection comes before cmd)?